modules/rgbd/src/hash_tsdf.cpp

```diff
@@ -289,17 +305,17 @@ inline TsdfType HashTSDFVolumeCPU::interpolateVoxel(const cv::Point3f& point) co
     float ty = point.y - iy;
     float tz = point.z - iz;
 
-    TsdfType vx[8];
+    float vx[8];
     for (int i = 0; i < 8; i++)
-        vx[i] = at(neighbourCoords[i] * voxelSize + point).tsdf;
+        vx[i] = tsdfToFloat(at(neighbourCoords[i] * voxelSize + point).tsdf);
```

Comment:
The function `at` performs several calls to hash table. Maybe manual inlining `at` + pushing hash table calls before the loop would give perf boost.

Reply:
I think there are 4 classes of points:
* Most popular case: all neighbors are inside 1 volume unit
* Neighbors share the same face of 2 volume units
* Neighbors share the same edge of 4 volume units
* Neighbors share the same vertex of 8 volume units
I don't know a flexible way to process all 4 cases, but if we cover at least the 1st one, a lot of hash table calls can be saved.